web/src/api/external/VectorDataManager.ts

```diff
@@ -0,0 +1,137 @@
+import { FireLayer } from 'api/external/fbaVectorSourceAPI'
+import axios, { AxiosInstance, AxiosResponse } from 'axios'
+import { IDBPDatabase, openDB } from 'idb'
+import { isEqual, isUndefined, uniq } from 'lodash'
+
+const fireCenterUrl =
+  'https://maps.gov.bc.ca/arcserver/rest/services/whse/bcgw_pub_whse_legal_admin_boundaries/MapServer/'
+
+/**
+ * Requests and caches vector data for fire centers/zones
+ */
+export class VectorDataManager {
+  private axiosInstance: AxiosInstance
+  private static dataStoreName = 'fireVectors'
+  private static fireCenterNameIdx = 'fireCenterNameIdx'
+  private db!: IDBPDatabase
+
+  constructor() {
+    this.axiosInstance = axios.create({
+      baseURL: fireCenterUrl
+    })
+  }
+
+  public async init(): Promise<void> {
+    this.db = await openDB('FireData', 1, {
+      upgrade(db) {
+        db.createObjectStore(VectorDataManager.dataStoreName, {
+          // Primary key, unique by layer and extent
+          keyPath: 'cacheKey'
+        })
+      }
+    })
+  }
+  public async getOrSet(
+    layer: FireLayer,
+    extent: number[]
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ): Promise<AxiosResponse<any> | undefined> {
+    const cachedValue = await this.db.get(
+      VectorDataManager.dataStoreName,
+      this.buildKey(layer, extent)
+    )
+    if (!isUndefined(cachedValue)) {
+      return cachedValue.data
+    }
+
+    try {
+      const { data } = await this.axiosInstance.get(this.buildUrl(layer, extent))
+      if (data.features.length > 0) {
+        const records = this.buildRecord(
+          layer,
+          extent,
+          this.buildKey(layer, extent),
+          data
+        )
+        const tx = this.db.transaction(VectorDataManager.dataStoreName, 'readwrite')
+        await Promise.all(records.map(record => tx.store.put(record)))
+      }
+      return data
+    } catch (error) {
+      console.log(error)
```

Comment:
is this all we want to do for catching errors? It doesn't really help users (I would expect most aren't going to know to inspect the console log). Is there something more helpful we can give as an error message to users?

Reply:
Yeah there's nothing we really can do if there are errors; users are not going to know what it means, and we have no control over fixing them.

Ideally we'd be sending stack traces to something like Sentry so we can track, reproduce, and fix errors we have control over, but we're not there yet: https://try.sentry-demo.com/organizations/correct-bird/issues/8322/?project=3013#exception

Also I'm hoping this code goes away soon.